test/integration/M3/Base.t.sol

```diff
@@ -0,0 +1,219 @@
+    ContractAddresses public contractAddresses;
+    ContractAddresses.ChainAddresses public chainAddresses;
+    ActorAddresses public actorAddresses;
+    ActorAddresses.Actors public actors;
+
+    // Rewards
+    RewardsReceiver public executionLayerReceiver;
+    RewardsReceiver public consensusLayerReceiver;
+    RewardsDistributor public rewardsDistributor;
+
+    // Staking
+    StakingNodesManager public stakingNodesManager;
+    StakingNode public stakingNodeImplementation;
+
+    // Assets
+    ynETH public yneth;
+
+    // Withdrawals
+    WithdrawalQueueManager public ynETHWithdrawalQueueManager;
+    ynETHRedemptionAssetsVault public ynETHRedemptionAssetsVaultInstance;
+
+    // EigenLayer
+    IEigenPodManager public eigenPodManager;
+    IDelegationManager public delegationManager;
+
+    // Mock Contracts to deploy
+    BeaconChainMock public beaconChain;
+
+    // Ethereum
+    IDepositContract public depositContractEth2;
+
+    address public constant GLOBAL_ADMIN = 0x72fdBD51085bDa5eEEd3b55D1a46E2e92f0837a5;
+
+    uint64 public constant GENESIS_TIME_LOCAL = 1 hours * 12;
+
+    function setUp() public virtual {
+        assignContracts();
+        upgradeYnToM3();
+    }
+
+    function assignContracts() internal {
+        contractAddresses = new ContractAddresses();
+        chainAddresses = contractAddresses.getChainAddresses(block.chainid);
+        actorAddresses = new ActorAddresses();
+        actors = actorAddresses.getActors(block.chainid);
+
+        // assign YieldNest addresses
+        {
+            yneth = ynETH(payable(chainAddresses.yn.YNETH_ADDRESS));
+            stakingNodesManager = StakingNodesManager(payable(chainAddresses.yn.STAKING_NODES_MANAGER_ADDRESS));
+            rewardsDistributor = RewardsDistributor(payable(chainAddresses.yn.REWARDS_DISTRIBUTOR_ADDRESS));
+            executionLayerReceiver = RewardsReceiver(payable(chainAddresses.yn.EXECUTION_LAYER_RECEIVER_ADDRESS));
+            consensusLayerReceiver = RewardsReceiver(payable(chainAddresses.yn.CONSENSUS_LAYER_RECEIVER_ADDRESS));
+        }
+
+        // assign Ethereum addresses
+        {
+            depositContractEth2 = IDepositContract(chainAddresses.ethereum.DEPOSIT_2_ADDRESS);
+        }
+
+        // assign Eigenlayer addresses
+        {
+            eigenPodManager = IEigenPodManager(chainAddresses.eigenlayer.EIGENPOD_MANAGER_ADDRESS);
+            delegationManager = IDelegationManager(chainAddresses.eigenlayer.DELEGATION_MANAGER_ADDRESS);
+        }
+
+        // deploy EigenLayer mocks
+        {
+            vm.warp(GENESIS_TIME_LOCAL);
+            beaconChain = new BeaconChainMock(EigenPodManager(address(eigenPodManager)), GENESIS_TIME_LOCAL);
+        }
+    }
+
+    function upgradeYnToM3() internal {
+        if (block.chainid != 17000) return;
+
+        // upgrade stakingNodesManager
+        {
+            vm.startPrank(ProxyAdmin(getTransparentUpgradeableProxyAdminAddress(address(stakingNodesManager))).owner());
```

Comment:
let's not prank this

prank the PROXY_ADMIN

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/159/commits/1069ccfd1cb7cedb873b964c87279e93206bd279